util/types/convert.go

```diff
@@ -164,6 +165,38 @@ func StrToUint(sc *variable.StatementContext, str string) (uint64, error) {
 	return uVal, errors.Trace(err)
 }
 
+// StrToDateTime convert str to MySQL DateTime.
+func StrToDateTime(str string, fsp int) (Time, error) {
+	return ParseTime(str, mysql.TypeDatetime, fsp)
+}
+
+// StrToDuration convert str to MySQL Time.
```

Comment:
yes, StrToDuration is ported from MySQL, but StrToDateTime is not, StrToDateTime reuse TiDB previous implement.